astropy/nddata/mixins/tests/test_ndarithmetic.py

```diff
@@ -1181,9 +1181,7 @@ def meta_fun_func(meta1, meta2, take="first"):
 
 def test_arithmetics_wcs_func():
     def wcs_comp_func(wcs1, wcs2, tolerance=0.1):
-        if tolerance < 0.01:
-            return False
-        return True
+        return tolerance >= 0.01
```

Comment:
Like the other comment, this is potentially changing the output type here if `tolerance` is a numpy array scalar. Maybe it doesn't matter but ruff changes should avoid subtle code changes, no?

Reply:
This one is semi manual anyway, ruff doesn't know how to simplify `not` VS inequalities. I can convert the output to a Python bool manually for better safety.
You're right that ruff should avoid this kind of changes manually but I think what it did here is right >99% of the time and it couldn't statically guess about the remaining epsilon% so in my honest opinion it's fine to keep this rule.

Reply:
This function is defined inside a test and is not even accessible from anywhere else. If the test still succeeds then the replacement was good enough. We don't need to worry about API stability here.

Reply:
This refactor was reverted entirely. Please consider closing this thread.